res/controllers/common-controller-scripts.js

```diff
@@ -152,10 +152,17 @@ script.midiDebug = function(channel, control, value, status, group) {
 // Returns the deck number of a "ChannelN" or "SamplerN" group
 script.deckFromGroup = function(group) {
     var deck = 0;
-    if (group.substring(2, 8) === "hannel") {
-        // Extract deck number from the group text
-        deck = group.substring(8, group.length - 1);
+    if (group.match("hannel.")) {
```

Comment:
The regular expression is matched at least once and in the first `if` arm even twice. I am still convinced that using a regular expressions for this purpose is overkill.

Reply:
Sorry, this was my mistake. It should have been removed now.